app/controllers/events/view/tickets/discount-codes/list.js

```diff
@@ -48,6 +48,23 @@ export default Controller.extend({
         .finally(() => {
           this.set('isLoading', false);
         });
+    },
+    toggleStatus(discountCode) {
+      this.set('isLoading', true);
+      discountCode.toggleProperty('isActive');
+      discountCode.save()
+        .then(() => {
+          this.notify.success(this.get('l10n').t('Discount Code has been updated successfully.'));
+        })
+        .catch(() => {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred.'));
```

Comment:
But @srv-twry there may be a case where the information the user has entered is causing the problem, shouldn't he be given time to check the form and see if he can change the incorrect details.

Reply:
This is just for toggling the status of the discount code from the list. There is no way there can be incorrect details.

Reply:
Ooh I thought this is for saving the form.

Reply:
It may be possible that server does not respond and timeout.

Reply:
What i'm saying is as follows: 
1. Assume that the discount code was "Active"
2. Now the status was toggled. So now the `discountCode.toggleProperty('isActive');` will run and toggle it. Now status is "Inactive"
3. Now you try to save it on the server but fails. Hence on the server you have the status still as "Active" while on the FE you will have the status as "Inactive". That's why I am asking you to change it back to "Active" in case the network request fails.